### ISSUE TITLE
[XML] Method with empty body marked as not covered, even though it is

### ISSUE THREAD

**Issue by IgnatBeresnev** [Kotlin/kotlinx-kover#259]
This issue might affect more cases than simply an empty body, but I haven't checked anything else.

Reproducer:
```kotlin
// main
class TestFile {

    fun foo() {
        println("should be covered")
    }

    fun bar() { }
}

// test
class TestFileTest {
    
    @Test
    fun shouldCheckFoo() {
        TestFile().foo()
    }

    @Test
    fun shouldCheckBar() {
        TestFile().bar()
    }
}
```

Results in the following HTML report:
![2022-10-20_11-19-07](https://user-images.githubusercontent.com/22685910/196909854-9ff5c250-cc72-4369-98b7-d0bf4d0de702.png)

But if you upload the XML report to codecov.io ([results here](https://app.codecov.io/github/IgnatBeresnev/kover-reproducers/blob/master/src/main/kotlin/org/jetbrains/kover/reproducer/KotlinFile.kt)), you'll see the following:
![2022-10-20_11-22-59](https://user-images.githubusercontent.com/22685910/196910037-fce63fc7-e5a7-48b1-8d8a-ed9c274e2d5b.png)

I don't think it's a bug of codecov, but rather of kover's xml report data. Codecov seems to calculate coverage based on source file coverage, and if we look at the xml report, we'll see the following:
```xml
        // IntelliJ results
        <sourcefile name="KotlinFile.kt">
            <line nr="3" mi="0" ci="2" mb="0" cb="0"/>
            <line nr="6" mi="0" ci="4" mb="0" cb="0"/>
  ---->     <line nr="9" mi="0" ci="0" mb="0" cb="0"/>
            <counter type="INSTRUCTION" missed="0" covered="6"/>
            <counter type="BRANCH" missed="0" covered="0"/>
            <counter type="LINE" missed="0" covered="3"/>
        </sourcefile>
```

Where line 9 (`bar()` declaration) is neither covered (`ci`) nor missed (`mi`). Moreover, it affects the `LINE` counter, which is 1 less than it should be (should be 4 covered instructions).

However, if you change the engine to Jacoco and generate reports for the same case, you'll see that everything is correct and that line 9 is covered (`ci=1`).

```xml
       // Jacoco results
        <sourcefile name="KotlinFile.kt">
            <line nr="3" mi="0" ci="3" mb="0" cb="0"/>
            <line nr="6" mi="0" ci="4" mb="0" cb="0"/>
            <line nr="7" mi="0" ci="1" mb="0" cb="0"/>
  ---->     <line nr="9" mi="0" ci="1" mb="0" cb="0"/>
            <counter type="INSTRUCTION" missed="0" covered="9"/>
            <counter type="LINE" missed="0" covered="4"/>
            <counter type="COMPLEXITY" missed="0" covered="3"/>
            <counter type="METHOD" missed="0" covered="3"/>
            <counter type="CLASS" missed="0" covered="1"/>
        </sourcefile>
```

Runnable reproducer can be found in [this repo](https://github.com/IgnatBeresnev/kover-reproducers), there are instructions on how to run it and upload the report. 

This was found while investigating #16 as it affects calculated coverage.

**Comment by zuevmaxim** [comment]
The issue is caused by agent ignoring empty lines if it is not the only line in a method. The agent also should count instructions in the same way

**Comment by zuevmaxim** [comment]
Fixed in agent version `1.0.688`

**Comment by shanshin** [comment]
To use custom engine you need to configure it in all projects where Kover is used

for Kotlin Gradle script
```
kover {
    engine.set(kotlinx.kover.api.IntellijEngine("1.0.688"))
}
```

for Groovy
```
kover {
    engine = kotlinx.kover.api.IntellijEngine("1.0.688")
}
```

**Comment by shanshin** [comment]
Fixed in `0.7.0-Alpha`
